app/views/users/show.html.haml

```diff
@@ -132,9 +132,36 @@
       - else
         %p{style: "padding-left: 7px;"} You haven't picked any ontologies yet
       
-        
-        %li= link_to("Subscribe","/users/subscribe/" + @user.username, class: "btn btn-sm btn-default")
-        %li= link_to("Unsubscribe","/users/un-subscribe/" + @user.email,class: "btn btn-sm btn-default")
+        %div
+          %h5{style:"margin-top:18px"} Subscription service 
+
+          %div{style:"margin-top:12px"}
+
+            %label.switch
+              %input{type: "checkbox", id: "subscribe", data: {url: '/toggle'}}
+              %span Subscribe to the #{$ANNOUNCE_LIST} user mailing list
+              %span.slider.round
+      
+            %br
+
+            %label.switch
+              %input{type: "checkbox", id: "unsubscribe", data: {url: '/toggle'}}
+              %span Unsubscribe from the #{$ANNOUNCE_LIST} user mailing list
+              %span.slider.round
+
+            
+            :javascript
+              document.getElementById('subscribe').addEventListener('change', function() {
+                if (this.checked) {
+                  window.location.href = "/users/subscribe/" + "#{@user.username}"
```

Comment:
Don't forget to protect  this two endpoints to be accessed only the owner and admins